packages/mui-joy/src/Radio/Radio.tsx

```diff
@@ -304,7 +304,7 @@ const Radio = React.forwardRef(function Radio(inProps, ref) {
   const overlay = inProps.overlay || radioGroup?.overlay || overlayProp;
 
   const radioChecked =
-    typeof checkedProp === 'undefined' && !!value
+    typeof checkedProp === 'undefined' && value !== undefined && value !== null
```

Comment:
Is empty string (`value=""`) valid then?

Reply:
True, I will update that.

Reply:
I have pushed and added that additional change.